core/field_dropdown.ts

```diff
@@ -114,30 +113,29 @@ export class FieldDropdown extends Field {
    * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
    */
   constructor(
-      menuGenerator: AnyDuringMigration[][]|Function|Sentinel,
-      opt_validator?: Function, opt_config?: FieldConfig) {
+      menuGenerator: MenuGenerator, opt_validator?: Function,
+      opt_config?: FieldConfig);
+  constructor(menuGenerator: Sentinel);
+  constructor(
```

Comment:
For function overloading in TS, the final function's parameters need to support all of the previous functions' parameters. The final constructor supports `MenuGenerator|Sentinel` since those are the cases that need to be handled in code within the constructor. It's also ultimately the limit of what can be provided, so a user *could* provide both a `menuGenerator: Sentinel` and a validator.

The first two constructors here just guide what combinations should be provided. Without providing any parameters, it'll suggest providing a `MenuGenerator`. If a `Sentinel` is provided, it'll suggest not providing a validator or config. Since Sentinel is only for subclasses, it seems like most users won't want to provide it.

Reply:
I think this combination of constructors makes sense!